targets/csharp/source/source/PluginManager.cs

```diff
@@ -25,9 +25,9 @@ private PluginManager()
         /// <param name="contract">The plugin contract.</param>
         /// <param name="instanceName">The optional plugin instance name. Instance names allow to have mulptiple plugins with the same contract.</param>
         /// <returns>The plugin instance.</returns>
-        public static IPlayFabPlugin GetPlugin(PluginContract contract, string instanceName = "")
+        public static T GetPlugin<T>(PluginContract contract, string instanceName = "") where T : IPlayFabPlugin
         {
-            return Instance.GetPluginInternal(contract, instanceName);
+            return (T)Instance.GetPluginInternal(contract, instanceName);
```

Comment:
Two options here, and I'm not sure either one is "right":
return (T)GetPluginInternal(); // Which will throw an exception if the cast fails.
return GetPluginInternal() as T; // Which will silently return null if the cast fails.
I could see an argument for either one, and I haven't thought through the customer experience to really have an opinion. But perhaps you should consider which option you think is best before landing.

Reply:
I thought about it too, and decided that explicit cast failure exception here is better than null ref exception somewhere else

---
In reply to: [202452655](https://github.com/PlayFab/SDKGenerator/pull/175#discussion_r202452655) [](ancestors = 202452655)